### DISCUSSION TITLE
Navigate to Multiple Scene

### DISCUSSION THREAD

**Discussion by salisuabubakar** [grahammendick/navigation#688] [Q&A] [answered]
Hi @grahammendick,
I've got another question here as part of my migration from Wix RNN to the Navigation Router. Am having challenges navigation from Scene A > B  > C.

Am able to scene stateNavigation.navigate('A')  to scene B but i cant stateNavigation.navigate('B') to scene C.All the scene are in the same stack.

example: 

const stateNavigatorHomeScreen = new StateNavigator([
  { key: 'homeScreen' },
  { key: 'watchmovDetScreen1', trackCrumbTrail: true },
  { key: 'fullMovieScreen1', trackCrumbTrail: true }
]);

        <TabBarItem title="Home" image={require('../../assets/images/home.png')} onPress={() => console.log("Home Navigated")}>
          <NavigationHandler stateNavigator={stateNavigatorHomeScreen}>
            <NavigationStack>
              <Scene
                stateKey="homeScreen"
                backgroundColor={() => Platform.OS === 'android' ? 'black' : 'white'}
                unmountStyle={from => from ? 'slide_in' : 'slide_out'}
              >
                <HomeScreen />
              </Scene>
              <Scene
                stateKey="watchmovDetScreen1"
                backgroundColor={() => Platform.OS === 'android' ? 'black' : 'white'}
                unmountStyle={from => from ? 'slide_in' : 'slide_out'}
                crumbStyle={from => from ? '' : 'empty'}
              >
                <WatchmovieDetailsScreen />
              </Scene>
              <Scene
                stateKey="fullMovieScreen1"
                backgroundColor={() => Platform.OS === 'android' ? 'black' : 'white'}
                unmountStyle={from => from ? 'slide_in' : 'slide_out'}
                crumbStyle={from => from ? '' : 'empty'}
              >
                <FullMovieScreen />
              </Scene>
            </NavigationStack>
          </NavigationHandler>
        </TabBarItem>

 Am able to navigate from homeScreen to watchmovDetScreen1 but i cant navigate from watchmovDetScreen1 to fullMovieScreen1.

Please help. 

Thanks

**Comment by grahammendick** [comment, marked as answer]
Hey, are you creating the `stateNavigatorHomeScreen` inside the render? Either move it out of the render or wrap it in a `useMemo`. If that's not the problem could you give me more info about what's going wrong, please?

**Reply by salisuabubakar** [reply]
Hi,
Thanks for your time.  Sorry for wasting your time on this. I had it working now. I forgot to add that scene to the same stack and use stateNavigator.nagive("") to navigate to the that scene. Its working now and am very happy. Thanks soo much. Am always greatfull
